### ISSUE TITLE
When this model can converge?

### ISSUE THREAD

**Issue by cnlinxi** [Tomiinek/Multilingual_Text_to_Speech#9]
I have trained this model at 7k steps, but I cannot get any acceptable audio. The dataset have 40h, and I predict linear spectrogram then use griffin-lim to get waveform.

**Comment by Tomiinek** [comment]
It depends on many factors and properties of your dataset.

I thing I used something like 50k training steps when training on the whole CSS10 dataset with ~150 hours, and something about 15k for a monolingual German subset of the CSS10 with ~15 training hours. Attention is usually established much faster (e.g., at half of the training). Be patient :wink:

If the model can not really establish the attention, I would recommend you playing with parameters associated with the guided attention (e.g., `guided_attention_gain`). The toleration of guided attention should not decay very fast or slow.

**Comment by cnlinxi** [comment]
Thank you for your answer and I'll try more. Great work, thank you for sharing.

**Comment by cnlinxi** [comment]
@Tomiinek Hi, I feel sad because I can't get intelligible speech at all. I train this model on TeslaP40 (24GB) and get the OOM error of GPU. So I set batch_size to 32. Or set batch_size to 16 and use gradient accumulation to back-propagation every 4 steps. Is this a bad idea? What did I do wrong?

**Comment by cnlinxi** [comment]
Can you give me a picture of the losses changes during training? During the training, the loss value of the model decreased to about 7300, and then it vibrated repeatedly. I think something I did wrong. Thank you.

**Comment by Tomiinek** [comment]
>  So I set batch_size to 32. Or set batch_size to 16 and use gradient accumulation to back-propagation every 4 steps.

I used batches such as 50 (with five or ten languages), but how many languages are you training on? Gradient accumulation good idea.

> Can you give me a picture of the losses changes during training?

Unfortunately, I do not have any training logs available now. Can you post some attention plots and the loss curves?

**Comment by Tomiinek** [comment]
And is the loss really that high? I would expect something about around 1, not 7300 :open_mouth:

**Comment by cnlinxi** [comment]
@Tomiinek here is the losses
<img width="747" alt="Snipaste_2020-08-03_15-38-06" src="https://user-images.githubusercontent.com/11946496/89157627-689f2d00-d59f-11ea-828a-0f1a638d2a34.png">

**Comment by cnlinxi** [comment]
and this is the alignment:
<img width="311" alt="Snipaste_2020-08-03_15-38-31" src="https://user-images.githubusercontent.com/11946496/89157743-94baae00-d59f-11ea-82b2-c234d5114641.png">

**Comment by Tomiinek** [comment]
The linear spectrograms have obviously weird magnitude, the loss of final spectrograms (`loss_mel_pos`) is too high, they should be normalized. How did you create them? Do you use `normalize_spectrogram = True`? At [this line](https://github.com/Tomiinek/Multilingual_Text_to_Speech/blob/e01fa7899b7570306df3e7bac72ced9958ca85e8/train.py#L56), they should have zero mean and unit variance (for each frequency channel).

**Comment by cnlinxi** [comment]
Thank you a lot. I will check this problem. If I have some progress, I will share it.

**Comment by Tomiinek** [comment]
This is my attention plot and output linear spectrogram after 500 training steps:

![individualImage3](https://user-images.githubusercontent.com/13215584/89169152-23b6ce80-d57e-11ea-872a-9a2ed1b0e701.png)

The total loss is around 0.6.

**Comment by cnlinxi** [comment]
Hi, I check the another model trained by another dataset. But I also get noise, it is so strange. I get the followed alignment and predicted linear spectrogram.

alignment:
<img width="283" alt="Snipaste_2020-08-03_18-08-21" src="https://user-images.githubusercontent.com/11946496/89172186-cc802080-d5b4-11ea-9014-ca61fed5b57c.png">

spectrogram:
<img width="707" alt="Snipaste_2020-08-03_18-09-21" src="https://user-images.githubusercontent.com/11946496/89172215-d86be280-d5b4-11ea-8478-18dc84e7204f.png">

target:
<img width="649" alt="Snipaste_2020-08-03_18-13-16" src="https://user-images.githubusercontent.com/11946496/89172277-f2a5c080-d5b4-11ea-84a2-7f65eaeca324.png">

**Comment by Tomiinek** [comment]
I do not understand :disappointed_relieved:  What is wrong with these plots?
